anvil-python/anvil/commands/juju.py

```diff
@@ -54,3 +61,47 @@ def run(self, status: Status | None) -> Result:
                 message="Could not find public ssh key (~/.ssh/id_rsa.pub)",
             )
         return Result(ResultType.COMPLETED)
+
+
+class RemoveJujuMachineStep(SunbeamRemoveJujuMachineStep):
+    def run(self, status: Status | None = None) -> Result:
+        res = super().run(status)
+        if res.result_type != ResultType.COMPLETED:
+            return res
+        try:
+            cmd = [
+                self._get_juju_binary(),
+                "wait-for",
+                "machine",
+                "-m",
+                CONTROLLER_MODEL,
+                str(self.machine_id),
+                "--query",
+                'life=="dead"',
+                "--timeout",
+                "5m",
```

Comment:
This timeout is running for `RemoveMachineUnitStep` per unit removal and after Sunbeam code has waited for the application to be in `active` state. So the waiting can be something short. The default for `wait-for` is 10 minutes and given the Juju limitation to detect that the unit is already removed before waiting (see comment in the code), I thought that 5 minutes per waiting time per unit is enough. We can always bump if we found at a later stage that this is too short.

Regarding the `RemoveJujuMachineStep`, given that we first remove all the units and only afterwards we remove the machine, the 5 minutes appear to be also enough. We can bump as well if we identify that this timeout is too short.

Reply:
So it seems that `maas-region` application unit needs 4 minutes to disappear. As such, I reverted the timeout set so that command can pick the default 10 minutes. You can observe this inside the logs of machine `maas-2.maas` removal. The first time the machine is removed without juju ha. The second time, I manually enabled juju-ha and then removed the machines again. Successful logs here: https://paste.ubuntu.com/p/jkhbrvWXRF/